src/liquid/components/ld-toggle/ld-toggle.tsx

```diff
@@ -0,0 +1,89 @@
+import '../../components' // type definitions for type checks and intelliSense
+import { Component, Element, h, Host, Prop } from '@stencil/core'
+import { JSXBase } from '@stencil/core/internal'
+import InputHTMLAttributes = JSXBase.InputHTMLAttributes
+import { cloneAttributes } from '../../utils/cloneAttributes'
+
+/**
+ * @virtualProp ref - reference to component
+ * @virtualProp {string | number} key - for tracking the node's identity when working with lists
+ */
+@Component({
+  tag: 'ld-toggle',
+  styleUrl: 'ld-toggle.css',
+  shadow: false,
+})
+export class LdToggle {
+  @Element() element: HTMLElement
+
+  /** Display mode. */
+  @Prop() mode?: 'small' | 'large'
+
+  /** Toggle tone. Use `'dark'` on white backgrounds. Default is a light tone. */
+  @Prop() tone: 'dark'
+
+  /** Disabled state of the toggle. */
+  @Prop() disabled = false
+
+  /** The input value. */
+  @Prop({ mutable: true, reflect: true }) checked: boolean
+
+  /** Set this property to `true` in order to mark the toggle visually as invalid. */
+  @Prop() invalid: false
+
+  private handleBlur(event) {
+    // Q: why timeout? is an explicit dispatch of events necessary? can't we somehow forward all events by default?
+    setTimeout(() => {
+      this.element.dispatchEvent(event)
+    })
+  }
+
+  private handleFocus(event) {
+    setTimeout(() => {
+      this.element.dispatchEvent(event)
+    })
+  }
+
+  private handleClick(event) {
+    // Q: why not dispatch like the other handlers? why aria-* instead of simply disabled?
+    if (event.target.getAttribute('aria-disabled') === 'true') {
+      event.preventDefault()
+      return
+    }
+    this.checked = event.target.getAttribute('aria-checked') !== 'true'
+  }
+
+  // TODO: memoized functions possible? simplify syntax, move to utils
+  private getClassNames = () => {
+    const classNames = [
+      'ld-toggle',
+      this.mode === 'large' ? 'ld-toggle--large' : 'ld-toggle--small',
+    ]
+
+    if (this.tone) classNames.push(`ld-toggle--${this.tone}`)
+    if (this.invalid) classNames.push('ld-toggle--invalid')
+
+    return classNames.join(' ')
+  }
+
+  render() {
+    return (
+      <Host>
+        <button
```

Comment:
I believe a11y tests will fail as the button doesn't have any text content nor an aria-label. Can you check?
I know that one would usually use a label with the toggle, but maybe we can give it an `aria-label="toggle"`? I would then definitely make sure that a screen reader still reads out what we expect. I would expect something like "toggle checked" in that case. But `aria-label` is usually kind of aggressive and may overwrite other things...

Reply:
Damn, I'm sorry. I just noticed that we need to use a checkbox implementation for the toggle as otherwise the CSS component doesn't make really much sense here.